### PR TITLE
Fix for GitLab Paginator in GitLab Provider

### PR DESCRIPTION
**Pull request by idavidov**
PR Type:
**Bug fix**

___
PR Description:
**This PR addresses an issue with the GitLab Provider where it was not correctly handling pagination for merge request diffs. The fix ensures that all diffs are retrieved by passing the 'get_all=True' argument to the 'list' method.**

___
PR Main Files Walkthrough:
-`git_providers/gitlab_provider.py`: The change is made in the '_set_merge_request' method where the 'get_all=True' argument is added to the 'list' method call on 'self.mr.diffs'. This ensures that all diffs are retrieved, not just the ones on the current page.

### CODE REVIEW

**Comment by idavidov** [comment]
/ask do we need last diff or all diffs?

**Comment by idavidov** [comment]
/ask then     self.last_diff = self.mr.diffs.list(order_by='updated_at', sort='desc', per_page=1)[0]
would be better solution?

**Comment by idavidov** [comment]
@okotek i still not 100% aware about most of details so i need your suggestion how better fix it
change which already in PR self.last_diff = self.mr.diffs.list(all=True)[-1]
or self.last_diff = self.mr.diffs.list(order_by='updated_at', sort='desc', per_page=1)[0]
for me AI in my 2 previous questions here explain well the difference but i have hard time to decide

File: pr_agent/git_providers/gitlab_provider.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def pr(self):
     def _set_merge_request(self, merge_request_url: str):
         self.id_project, self.id_mr = self._parse_merge_request_url(merge_request_url)
         self.mr = self._get_merge_request()
-        self.last_diff = self.mr.diffs.list()[-1]
+        self.last_diff = self.mr.diffs.list(get_all=True)[-1]
```
Comment (okotek) [inline pr_agent/git_providers/gitlab_provider.py:52]:
```suggestion
        diffs = self.mr.diffs.list(get_all=True)
        self.last_diff = diffs[-1] if diffs else None
```

**Comment by okotek** [comment]
I'd go with get_all, let's merge

**Review by okotek — APPROVED** [review]

**Pull request merged**